### ISSUE TITLE
unexpected behavior with ATL transforms

### ISSUE THREAD

**Issue by the66F95** [renpy/renpy#4942]
if your ATL transformation block has arguments and they are used, you will see a quite erratic behavior compared to before deep_compare() was introduced in commit 468733bd75ab650008d918804f741dfa88661f29 and commit b2a49cc60eeaf614ac1faf5ad806865989bb24b9.

for example, execute the attached script with a current nightly and 7.6.1/8.1.1.

[script.zip](https://github.com/renpy/renpy/files/12458028/script.zip)

**Comment by qTich** [comment]
The behavior for variables has been changed there, to use the behavior before these changes, you need to replace the variable `config.at_transform_compare_full_context = True`. [doc](https://www.renpy.org/doc/html/incompatible.html) - `At Transform and Global Variables`

**Comment by the66F95** [comment]
`config.at_transform_compare_full_context = True` is no longer used, was removed in one of the referenced commits

`At Transform and Global Variables` been there, done that
```py
screen atl_test_menu4(items):
    $ p = pos2

    frame at position(p):
        xpadding 50

        vbox:
            for i in items:
                textbutton i.caption action i.action```

**Comment by the66F95** [comment]
the 3rd screen works because `pos2` is declared by a `define` and should therefore be constant.
the next screen uses a changed `pos2` and already doesn't work anymore, even if the global `pos2` is casted into a local variable of the screen.
the 5th screen works again because the not "constant" `pos3` is used directly and not as an argument.

tl;dr
as far as i can see, currently only "constant" values can be used as a transformation argument.
but that renders the concept of arguments somewhat ineffective. 🤭
